internal/providers/terraform/azure/registry.go

```diff
@@ -166,8 +174,35 @@ var FreeResources []string = []string{
 	"azurerm_cosmosdb_sql_user_defined_function",
 
 	// Azure DNS
+	"azurerm_dns_a_record",
+	"azurerm_dns_aaaa_record",
+	"azurerm_dns_caa_record",
+	"azurerm_dns_cname_record",
+	"azurerm_dns_mx_record",
+	"azurerm_dns_ns_record",
+	"azurerm_dns_ptr_record",
+	"azurerm_dns_srv_record",
+	"azurerm_dns_txt_record",
+	"azurerm_private_dns_a_record",
+	"azurerm_private_dns_a_record",
+	"azurerm_private_dns_cname_record",
+	"azurerm_private_dns_mx_record",
+	"azurerm_private_dns_ptr_record",
+	"azurerm_private_dns_srv_record",
+	"azurerm_private_dns_txtrecord",
```

Comment:
```suggestion

```

We support these as paid resources by showing the cost for the queries made to them, e.g.:
```
 azurerm_dns_a_record.withoutUsage                                                       
 └─ DNS queries (first 1B)          Monthly cost depends on usage: $0.40 per 1M queries  
```

Reply:
Hi,
thanks for the feedback. I would've thought that it is only included in the DNS Zone itself since MS is only charging per zones and per x million requests, which is independent from the number of records that the DNS zone has.

Reply:
good point! we debated about it at the time and went with showing that cost component in the DNS records as the zone might be defined in another project or done outside of infra-as-code -- however, we could also show it in the DNS zone. That way users can decide where to [model the usage](https://www.infracost.io/docs/usage_based_resources/#infracost-usage-file), in the DNS zone or record, or both (so they're summed).

Reply:
Looking at it from a larger scale it might make sense to only show this in the DNS Zone, especially if companies have a large number of DNS Records (>100), this could clutter up the output. 
Again I'm not 100% if the output in infracost would only show the resource type once or repeatedly for every record separately?

Reply:
it would repeat for every resource, so as you suggested, it's probably best if we change that to roll it up under the DNS Zone.